tests/judges/test_instruction_judge.py

```diff
@@ -0,0 +1,298 @@
+        result = judge.evaluate(inputs, outputs)
+
+        # Verify canonical parsing was used
+        mock_parse_inputs.assert_called_once_with(inputs)
+        mock_parse_outputs.assert_called_once_with(outputs)
+        mock_judge.assert_called_once_with(
+            inputs={'request': 'parsed request'}, 
+            outputs={'response': 'parsed response'}
+        )
+        
+        # Verify version metadata was added
+        self.assertEqual(result.metadata, {'version': '1'})
+
+    @patch('server.judges.instruction_judge.make_judge')
+    def test_register_scorer_with_experiment(self, mock_make_judge):
+        """Test register_scorer with experiment ID."""
+        mock_judge = Mock()
+        mock_registered = Mock()
+        mock_judge.register.return_value = mock_registered
+        mock_make_judge.return_value = mock_judge
+        
+        judge = InstructionJudge(
+            name='Register Test Judge',
+            user_instructions='Test registration',
+            experiment_id='exp123',
+        )
+
+        result = judge.register_scorer()
+
+        # Verify register was called with proper naming and experiment_id
+        mock_judge.register.assert_called_once()
+        call_args = mock_judge.register.call_args
+        self.assertIn('name', call_args.kwargs)
+        self.assertEqual(call_args.kwargs['experiment_id'], 'exp123')
+        self.assertEqual(result, mock_registered)
+
+    @patch('server.judges.instruction_judge.make_judge')
+    def test_register_scorer_without_experiment(self, mock_make_judge):
+        """Test register_scorer without experiment ID."""
+        mock_judge = Mock()
+        mock_registered = Mock()
+        mock_judge.register.return_value = mock_registered
+        mock_make_judge.return_value = mock_judge
+        
+        judge = InstructionJudge(name='No Exp Judge', user_instructions='Test without experiment')
+
+        result = judge.register_scorer()
+
+        # Verify register was called without experiment_id
+        mock_judge.register.assert_called_once()
+        call_args = mock_judge.register.call_args
+        self.assertIn('name', call_args.kwargs)
+        self.assertNotIn('experiment_id', call_args.kwargs)
+
+    @patch('server.judges.instruction_judge.make_judge')
+    @patch('server.judges.instruction_judge.logger')
+    def test_register_scorer_error_handling(self, mock_logger, mock_make_judge):
+        """Test register_scorer handles errors gracefully."""
+        mock_judge = Mock()
+        mock_judge.register.side_effect = Exception("Registration failed")
+        mock_make_judge.return_value = mock_judge
+        
+        judge = InstructionJudge(
+            name='Error Register Judge', user_instructions='This registration will fail'
+        )
+
+        result = judge.register_scorer()
+
+        # Should return None and log warning
+        self.assertIsNone(result)
+        mock_logger.warning.assert_called_once()
```

Comment:
Same comment as below -- this assertion is a weak signal that proper error handling was performed.